src/cli/mod.rs

```diff
@@ -61,6 +84,37 @@ impl CliApp {
 
         Ok(())
     }
+
+    #[cfg(feature = "flightsql")]
+    async fn flightsql_execute_files(&self, files: Vec<PathBuf>) -> color_eyre::Result<()> {
+        info!("Executing files: {:?}", files);
+        for file in files {
+            let file = std::fs::read_to_string(file)?;
+            self.exec_from_flightsql(file).await?;
+        }
+
+        Ok(())
+    }
+
+    #[cfg(feature = "flightsql")]
+    async fn exec_from_flightsql(&self, sql: String) -> color_eyre::Result<()> {
+        let client = self.execution.flightsql_client();
```

Comment:
What I was hoping was that `self.execution` held the difference and the CLI didn't have to worry about how a query was executed -- instead it would configure `self.execution`  differently depending on if the flightsql feature was enabled

Reply:
Yes makes sense.  Something to note is that for the TUI both the local and FlightSQL interfaces are needed (i.e. not one or the other like with the CLI).  Further, I had in mind adding a 3rd client for DataFusion-Ray.

I envisioned working towards something like the below eventually but wanted to see some features land first (i.e. this and pagination for local and FlightSQL) before to make sure I had the use case right first.  If you already have the structure in mind though of course we can start working towards that already.

```
enum ExecutionClient {
    Local,
    FlightSQL,
    Ray,
}

struct ExecutionRequest {
    client: ExecutionClient,
    sql: String
}

impl ExecutionContext {
    fn execute(&self, request: ExecutionRequest) -> Result<Vec<RecordBatch>>

    async fn create_stream<S, E>(&self, request: ExecutionRequest) -> Result<S>
    where
        S: Stream<Item = Result<RecordBatch, E>> + Unpin,
        E: Error,
}
```